### PR TITLE
Habit Controller

### PR DESCRIPTION
**Pull request by desmondj2atwit**

### CODE REVIEW

File: lib/controllers/habit_controller.dart [hunk 1]
```diff
@@ -0,0 +1,325 @@
+import 'dart:ffi';
+import 'package:flutter/cupertino.dart';
+import 'package:intl/intl.dart';
+import 'package:spark/postgres_functions.dart';
+
+//habits: habit_id, user_id, title, note, start_date, end_date, frequency, reminders, reminder_message, target_type, category, quantity
+//activities: user_id, habit_id, timestamp, quanity
+
+//Homepage methods
```
Comment (lyjohnson) [inline lib/controllers/habit_controller.dart]: I don't see a class for "HabitController", are these all just floating methods? These should be wrapped in a class that implements "ChangeNotifier"

File: lib/controllers/habit_controller.dart [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+  return tomorrowsHabitsQuickView;
+}
+
+//New Habit page
+Future<String> createNewHabit(
+    String userID,
+    String title,
+    String note,
+    DateTime start,
+    DateTime end,
```
Comment (lyjohnson) [inline lib/controllers/habit_controller.dart]: I think "end" time should be nullable for habits that repeat forever

**Review by lyjohnson — COMMENTED** [review]
We should work to remove the "dynamic" type if we can. It can make the code a bit unstable.

**Comment by desmondj2atwit** [comment]
> We should work to remove the "dynamic" type if we can. It can make the code a bit unstable.

@lyjohnson Good point, I'll go into the query functions and add checks that it's getting proper data types from the database. As for the controller, half of the dynamics are from the maps that methods return. If the datatype is checked before the return is that okay or should I find a new way to return the information?

**Pull request merged**